python/main.py

```diff
@@ -1,45 +1,157 @@
+    conn = get_db_connection()
+    # カテゴリが存在するか確認し、なければ追加
+    category_id = conn.execute('SELECT id FROM categories WHERE name = ?', (category,)).fetchone()
+    if category_id is None:
+        conn.execute('INSERT INTO categories (name) VALUES (?)', (category,))
+        category_id = conn.execute('SELECT last_insert_rowid()').fetchone()[0]
+    else:
+        category_id = category_id['id']
+    
+    # 画像の処理
+    image_contents = await image.read()
+    image_hash = hashlib.sha256(image_contents).hexdigest()
+    image_filename = f"{image_hash}.jpg"
+    with open(images / image_filename, 'wb') as file:
+        file.write(image_contents)
+    
+    # 商品情報をデータベースに保存
+    conn.execute('INSERT INTO items (name, category_id, image_name) VALUES (?, ?, ?)',
+                 (name, category_id, image_filename))
+    conn.commit()
+    conn.close()
+    return {"message": f"Item received: {name}", "category": category, "image_name": image_filename}
+
+@app.get("/search")
+def search_items(keyword: str):
+    conn = get_db_connection()
+    items = conn.execute('''SELECT items.name, categories.name AS category, items.image_name
+                            FROM items
+                            JOIN categories ON items.category_id = categories.id
+                            WHERE items.name LIKE ?''', ('%' + keyword + '%',)).fetchall()
+    conn.close()
+    return {"items": [dict(item) for item in items]}
+
+    
+    # # 画像のファイル名の取得
+    # image_contents = await image.read()
+    # image_hash = hashlib.sha256(image_contents).hexdigest()
+    # image_filename = f"{image_hash}.jpg"
+    # with open(images / image_filename, 'wb') as file:
+    #     file.write(image_contents)
+
+    # # 新しい商品をJSONに追加
+    # new_item = {"name":name, "category":category, "image_name":image_filename}
+    # with open(file_path, 'r+') as file:
+    #     items_data = json.load(file)
+    #     items_data.append(new_item)
+    #     file.seek(0)
+    #     json.dump(items_data, file, indent=4)
+
 
+    logger.info(f"Receive item: {name}, {category}")
+    return {"message": f"item received: {name}, {category}"} # ちょっと形式違うけどこれでいいのかな？
 
-@app.get("/image/{image_name}")
-async def get_image(image_name):
+
+
+@app.get("/image/{image_filename}")
+async def get_image(image_filename: str):
     # Create image path
-    image = images / image_name
+    image_path = images / image_filename
 
-    if not image_name.endswith(".jpg"):
+    if not image_filename.endswith(".jpg"):
         raise HTTPException(status_code=400, detail="Image path does not end with .jpg")
 
-    if not image.exists():
-        logger.debug(f"Image not found: {image}")
+    if not image_path.exists():
+        logger.debug(f"Image not found: {image_filename}")
         image = images / "default.jpg"
 
     return FileResponse(image)
+
+
+# データベース接続用の関数
+def get_db_connection() -> Connection:
+    conn = sqlite3.connect('db/mercari.sqlite3')
```

Comment:
今回pythonディレクトリの下にmercari.sqlite3を作成しているので、ここのパスは修正する必要があります。
```suggestion
    conn = sqlite3.connect('mercari.sqlite3')
```